### ISSUE TITLE
Image files on set pages with dashes in filename don't work

### ISSUE THREAD

**Issue by wickr** [OregonDigital/oregondigital#583]
When images on a set page have a dash in the filename, they don't load due to a routing error.

`No route matches [GET] "/assets/sets/oimb/010-6872-11212011181502.jpg"`

Workaround is to switch dashes to an underscore.

**Comment by terrellt** [comment]
@wickr Is this still a bug? Do we have an example?

**Comment by wickr** [comment]
@terrellt Yeah. No active example since workarounds were done. If you take any set page with an image, view just that image in the browser, add a dash into the filename, you'll get the routing error before file not found error.

**Comment by srabun** [comment]
use _ instead of - for now. but why is it broken?

**Comment by jechols** [comment]
Rails bug: https://github.com/rails/rails/issues/9788 - We're just lucky enough to have this precise kind of issue.  Two workarounds: don't use hyphen *or* shorten the filename.

**Comment by jechols** [comment]
Er, sprockets bug - Rails just happens to rely on sprockets

**Comment by jechols** [comment]
(Note that this could bite us on any asset file, not just images - css, fonts, JS, etc)

**Comment by srabun** [comment]
might need to update rails....

**Comment by gorn** [comment]
I can confirm that on rails 5.1.6.1 this is still an issue. This time images with + in the name are not server. Dashes work. How is that even possible?
